pulp_container/tests/functional/api/test_rbac_repo_versions.py

```diff
@@ -0,0 +1,269 @@
+        cls.user_creator["remote_api"].delete(cls.remote.pulp_href)
+        delete_orphans()
+        del_user(cls.user_creator)
+        del_user(cls.user_repov_remover)
+        del_user(cls.user_repo_remover)
+        del_user(cls.user_reader)
+        del_user(cls.user_helpless)
+
+    def test_repov_list(self):
+        """
+        Test that users can list repository versions if they have enough rights
+        """
+        self.assertEqual(self.repo_version_api.list(self.repository.pulp_href).count, 2)
+        self.assertEqual(
+            self.user_creator["repo_version_api"].list(self.repository.pulp_href).count, 2
+        )
+        self.assertEqual(
+            self.user_reader["repo_version_api"].list(self.repository.pulp_href).count, 2
+        )
+        with self.assertRaises(ApiException):
+            self.user_helpless["repo_version_api"].list(self.repository.pulp_href)
+
+    def test_repov_read(self):
+        """
+        Test that users can read specific repository versions if they have enough rights
+        """
+        repository = self.repositories_api.read(self.repository.pulp_href)
+        self.repo_version_api.read(repository.latest_version_href)
+        self.user_creator["repo_version_api"].read(repository.latest_version_href)
+        self.user_reader["repo_version_api"].read(repository.latest_version_href)
+        with self.assertRaises(ApiException):
+            self.user_helpless["repo_version_api"].read(repository.latest_version_href)
+
+    def test_repov_delete(self):
+        """
+        Test that users can delete repository versions if they have enough rights
+        """
+
+        def create_new_repo_version():
+            """
+            Create a new repo version to delete it later by a test user
+            """
+            manifest_a = self.get_manifest_by_tag("manifest_a")
+            self.tag_image(manifest_a, "new_tag")
+            repository = self.repositories_api.read(self.repository.pulp_href)
+            return repository.latest_version_href
+
+        repository = self.repositories_api.read(self.repository.pulp_href)
+        with self.assertRaises(ApiException):
+            self.user_helpless["repo_version_api"].delete(repository.latest_version_href)
+        with self.assertRaises(ApiException):
+            self.user_reader["repo_version_api"].delete(repository.latest_version_href)
+
+        response = self.repo_version_api.delete(create_new_repo_version())
+        monitor_task(response.task)
+
+        response = self.user_creator["repo_version_api"].delete(create_new_repo_version())
+        monitor_task(response.task)
+
+        response = self.user_repov_remover["repo_version_api"].delete(create_new_repo_version())
+        monitor_task(response.task)
+
+        response = self.user_repo_remover["repo_version_api"].delete(create_new_repo_version())
+        monitor_task(response.task)
+
+
+class PushRepoVersionTestCase(unittest.TestCase):
```

Comment:
I wonder if I should move push tests to the module where push content tests are, to reduce the setup time for tests.

Reply:
push content tests are already quite long, i believe eventually we goona split those.
My personal preference would be to keep tests organized by 'topic' even though we are going to sacrifice on the setup time.